src/solvers/strings/string_constraint.cpp

```diff
@@ -22,7 +22,7 @@ static bool cannot_be_neg(const exprt &expr)
   satcheck_no_simplifiert sat_check(null_message_handler);
   symbol_tablet symbol_table;
   namespacet ns(symbol_table);
-  boolbvt solver(ns, sat_check);
+  boolbvt solver(ns, sat_check, null_message_handler);
```

Comment:
Is `null_message_handler` really the best option here?

Reply:
1) This is an internal call to check consistency of some subexpressions, so I think it's ok. (The same rationale that I had used when inserting `null_message_handler` a few lines above.)
2) Using a different message handler yields a diff of 400+ lines, because a lot of procedures in the string solver are free functions, and therefore almost each and every procedure needs to take an additional parameter.

Reply:
Yikes!  This is a bit concerning.  Is this because of the "free function" approach or simply the way it has been used?

Reply:
The free-function approach applied to a lot tightly coupled functions.